app/src/main/kotlin/com/saltedge/authenticator/features/consents/ConsentsListViewModel.kt

```diff
@@ -0,0 +1,179 @@
+import com.saltedge.authenticator.features.connections.common.ConnectionViewModel
+import com.saltedge.authenticator.features.connections.list.collectConnectionViewModel
+import com.saltedge.authenticator.features.consents.common.ConsentItemViewModel
+import com.saltedge.authenticator.models.Connection
+import com.saltedge.authenticator.models.ViewModelEvent
+import com.saltedge.authenticator.models.repository.ConnectionsRepositoryAbs
+import com.saltedge.authenticator.sdk.AuthenticatorApiManagerAbs
+import com.saltedge.authenticator.sdk.contract.FetchEncryptedDataListener
+import com.saltedge.authenticator.sdk.model.ConnectionID
+import com.saltedge.authenticator.sdk.model.ConsentData
+import com.saltedge.authenticator.sdk.model.EncryptedData
+import com.saltedge.authenticator.sdk.model.connection.ConnectionAndKey
+import com.saltedge.authenticator.sdk.model.error.ApiErrorData
+import com.saltedge.authenticator.sdk.tools.crypt.CryptoToolsAbs
+import com.saltedge.authenticator.sdk.tools.keystore.KeyStoreManagerAbs
+import com.saltedge.authenticator.tools.daysTillExpire
+import kotlinx.coroutines.*
+import kotlin.coroutines.CoroutineContext
+
+class ConsentsListViewModel(
+    private val appContext: Context,
+    private val connectionsRepository: ConnectionsRepositoryAbs,
+    private val keyStoreManager: KeyStoreManagerAbs,
+    private val apiManager: AuthenticatorApiManagerAbs,
+    private val cryptoTools: CryptoToolsAbs
+) : ViewModel(), LifecycleObserver, FetchEncryptedDataListener, CoroutineScope {
+
+    private val decryptJob: Job = Job()
+    override val coroutineContext: CoroutineContext
+        get() = decryptJob + Dispatchers.IO
+
+    val listItems = MutableLiveData<List<ConsentItemViewModel>>()
+    val connectionItem = MutableLiveData<ConnectionViewModel>()
+
+    private var connection = Connection()
+    private var connectionGuid = ""
+    private var connectionsAndKeys: Map<ConnectionID, ConnectionAndKey> =
+        collectConnectionsAndKeys(
+            connectionsRepository,
+            keyStoreManager
+        )
+    var onListItemClickEvent = MutableLiveData<ViewModelEvent<Int>>()
+        private set
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_START)
+    fun onStart() {
+        updateViewsContent()
+        refreshConsents()
+    }
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onDestroy() {
+        decryptJob.cancel()
+    }
+
+    override fun onFetchEncryptedDataResult(
+        result: List<EncryptedData>,
+        errors: List<ApiErrorData>
+    ) {
+        processOfEncryptedConsentsResult(encryptedList = result)
+    }
+
+    fun setInitialData(connectionGuid: String?, consents: List<ConsentData>) {
+        listItems.postValue(
+            listOf(
+                ConsentItemViewModel(
+                    id = "id",
+                    tppName = "Example Dashboard",
+                    consentTypeDescription = "aisp",
```

Comment:
use human readable description from `strings`

Reply:
This is the hard code that was used for displaying, nothing more.